File: src/app/src/main/java/jp/co/e2/dogage/dialog/PhotoDialog.java
package jp.co.e2.dogage.dialog;

import jp.co.e2.dogage.R;
import jp.co.e2.dogage.dialog.PhotoDialog.CallbackListener;

import android.support.v7.app.AlertDialog;
import android.app.Dialog;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

/**
 * 画像表示ダイアログ
 */
public class PhotoDialog extends BaseDialog<CallbackListener> {
    private static final String PARAM_BITMAP = "bitmap";

    /**
     * ファクトリーメソッド
     *
     * @param bitmap ビットマップ
     * @return SampleDialog
     */
    public static PhotoDialog newInstance(Bitmap bitmap) {
        PhotoDialog dialog = new PhotoDialog();

        Bundle bundle = new Bundle();
        bundle.putParcelable(PARAM_BITMAP, bitmap);
        dialog.setArguments(bundle);

        return dialog;
    }

    /**
     * ${inheritDoc}
     */
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        View view = getActivity().getLayoutInflater().inflate(R.layout.dialog_photo, null);

        ImageView imageViewPhoto = view.findViewById(R.id.imageViewPhoto);
        imageViewPhoto.setImageBitmap((Bitmap) getArguments().getParcelable(PARAM_BITMAP));

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setView(view);

        return builder.create();
    }

    /**
     * コールバックリスナー
     */
    interface CallbackListener {
    }
}

File: src/app/src/main/java/jp/co/e2/dogage/dialog/PhotoDialog.kt
package jp.co.e2.dogage.dialog

import jp.co.e2.dogage.R
import jp.co.e2.dogage.dialog.PhotoDialog.CallbackListener

import android.app.Dialog
import android.graphics.Bitmap
import android.net.Uri
import android.os.Bundle
import android.os.Parcelable
import android.widget.ImageView
import androidx.appcompat.app.AlertDialog
import com.squareup.picasso.Picasso
import jp.co.e2.dogage.common.AndroidUtils
import jp.co.e2.dogage.config.Config

/**
 * 画像表示ダイアログ
 */
class PhotoDialog : BaseDialog<CallbackListener>() {

    companion object {
        private const val PARAM_URI = "uri"

        /**
         * ファクトリーメソッド
         *
         * @param uri URI
         * @return SampleDialog
         */
        fun newInstance(uri: Uri): PhotoDialog {
            val dialog = PhotoDialog()

            val bundle = Bundle()
            bundle.putParcelable(PARAM_URI, uri)
            dialog.arguments = bundle

            return dialog
        }
    }

    /**
     * ${inheritDoc}
     */
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val view = activity!!.layoutInflater.inflate(R.layout.dialog_photo, null)

        val imageViewPhoto = view.findViewById<ImageView>(R.id.imageViewPhoto)

        val uri = arguments!!.getParcelable<Parcelable>(PARAM_URI) as Uri;
        val size = AndroidUtils.dpToPixel(activity, Config.PHOTO_BIG_DP)
        Picasso.get().load(uri).resize(size, size).centerCrop().into(imageViewPhoto)

        val builder = AlertDialog.Builder(context!!, R.style.TransparentDialogStyle)
        builder.setView(view)

        return builder.create()
    }

    /**
     * コールバックリスナー
     */
    interface CallbackListener
}
